feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -272,8 +279,13 @@ bool target_radio_config_set2(const struct schema_Wifi_Radio_Config *rconf,
 	if (changed->tx_power)
 		blobmsg_add_u32(&b, "txpower", rconf->tx_power);
 
-	if (changed->tx_chainmask)
-		blobmsg_add_u32(&b, "tx_antenna", rconf->tx_chainmask);
+	if (changed->tx_chainmask) {
```

Comment:
> In radio_nl80211.c/nl80211_add_phy() we capture the tx_ant_avail and rx_ant_avail. We should add some code to make sure that we do no not request more antennas than we actually have.
> 
> if ((tx_chainmask & tx_ant_avail) != tx_chainmask)) then NOTICE level log and use the tx_ant_avail.
> Same for RX.

Done, Thank you

Reply:
I don't think your solution is correct.  When they are not equal, then you should explicitly apply the max available.

This covers the case where a 2x2 available was configured as 1x1 and then reconfigured as 3x3.  It should do 2x2 because closer to request.
Note: eventually this should be moot, if we can tell the cloud our actual limits and then they apply them.

Reply:
> I don't think your solution is correct. When they are not equal, then you should explicitly apply the max available.
> 
> This covers the case where a 2x2 available was configured as 1x1 and then reconfigured as 3x3. It should do 2x2 because closer to request.
> Note: eventually this should be moot, if we can tell the cloud our actual limits and then they apply them.

got it, I'll do that